src/sage/graphs/matching_covered_graph.py

```diff
@@ -0,0 +1,1228 @@
+                    try:
+                        u, v = u
+                    except Exception:
+                        pass
+
+        else:
+            if v is None:
+                try:
+                    u, v = u
+                except Exception:
+                    pass
+
+        if u in self and v in self:
+            if u == v:
+                raise ValueError('loops are not allowed in '
+                                 'matching covered graphs')
+            G = Graph(self)
+            G.add_edge(u, v, label=label)
+
+            try:
+                self.__init__(data=G, matching=self.get_matching())
+            except Exception:
+                raise ValueError('the graph obtained after the addition of '
+                                 'edge (%s) is not matching covered'
+                                 % str((u, v, label)))
+
+        else:
+            # At least one of u or v is a nonexistent vertex.
+            # Thus, the resulting graph is either disconnected
+            # or has an odd order, hence not matching covered
+            raise ValueError('the graph obtained after the addition of edge '
+                             '(%s) is not matching covered'
+                             % str((u, v, label)))
+
+    def add_vertex(self, name=None):
+        r"""
+        Add a vertex to the (matching covered) graph.
+
+        .. NOTE::
+
+            This method overwrites the
+            :meth:`~sage.graphs.generic_graph.GenericGraph.add_vertex` method
+            to ensure that isolated vertices are forbidden in
+            :class:`~MatchingCoveredGraph`.
+
+        INPUT:
+
+        - ``name`` -- an immutable object (default: ``None``); when no name is
+          specified (default), then the new vertex will be represented by the
+          least integer not already representing a vertex. ``name`` must be an
+          immutable object (e.g., an integer, a tuple, etc.).
+
+        OUTPUT:
+
+        - If ``name`` specifies an existing vertex, then nothing is done.
+          Otherwise a :exc:`ValueError` is returned with no change to the
+          existing (matching covered) graph is returned since matching covered
+          graphs are free of isolated vertices.
+
+        EXAMPLES:
+
+        Adding an existing vertex::
+
+            sage: P = graphs.PetersenGraph()
+            sage: P
+            Petersen graph: Graph on 10 vertices
+            sage: G = MatchingCoveredGraph(P)
+            sage: G
+            Matching covered petersen graph: graph on 10 vertices
+            sage: # needs random
+            sage: u = random.choice(list(G))
+            sage: G.add_vertex(u)
+            sage: G
+            Matching covered petersen graph: graph on 10 vertices
+
+        Adding a new/ non-existing vertex::
+
+            sage: P = graphs.PetersenGraph()
+            sage: P
+            Petersen graph: Graph on 10 vertices
+            sage: G = MatchingCoveredGraph(P)
```

Comment:
here you can reuse the graph G created in the previous block.

Reply:
But, we have a comment followed by a `::` just before that. Won't that result in a different code box in the documentation UI?

Reply:
yes, it creates a new code box, but this is something we do frequently to have a lighter, yet meaningful, documentation